src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -763,16 +763,17 @@ private void RegisterApplicationLifetimeEvents()
             _applicationLifetime.ApplicationStopping.Register(() =>
             {
                 Interlocked.Exchange(ref _applicationStopping, 1);
-                if (_environment.DrainOnApplicationStoppingEnabled())
+                var drainModeManager = _host?.Services.GetService<IDrainModeManager>();
+                if (drainModeManager != null)
                 {
-                    var drainModeManager = _host?.Services.GetService<IDrainModeManager>();
-                    if (drainModeManager != null)
-                    {
-                        _logger.LogDebug("Application Stopping: initiate drain mode");
-                        drainModeManager.EnableDrainModeAsync(CancellationToken.None);
-                        // Workaround until https://github.com/Azure/azure-functions-host/issues/7188 is addressed.
-                        Thread.Sleep(TimeSpan.FromMinutes(10));
-                    }
+                    _logger.LogDebug("Application Stopping: Initiate drain mode.");
+                    drainModeManager.EnableDrainModeAsync(CancellationToken.None);
```

Comment:
So it looks like we init drain even if `DrainOnApplicationStoppingEnabled` is not enabled, so do we need the check below? 

Just trying to understand why we're doing this 😄

Reply:
Trying to answer why shouldn't we stop listeners as part of application shutdown :) - this will avoid calling DrainMode api from the platform as part of worker unassignment flow